src/libraries/System.Security.Cryptography.Encoding/tests/Cbor.Tests/CborReaderTests.Tag.cs

```diff
@@ -239,15 +239,31 @@ public static void ReadDateTimeOffset_InvalidFormat_ShouldRollbackToInitialState
 
             reader.ReadStartArray();
             int bytesRead = reader.BytesRead;
-            int bytesRemaining = reader.BytesRemaining;
             Assert.Throws<FormatException>(() => reader.ReadDateTimeOffset());
 
             Assert.Equal(bytesRead, reader.BytesRead);
-            Assert.Equal(bytesRemaining, reader.BytesRemaining);
             Assert.Equal(CborReaderState.Tag, reader.PeekState());
             Assert.Equal(CborTag.DateTimeString, reader.ReadTag());
         }
 
+        [Fact]
+        public static void ReadDateTimeOffset_StrictConformance_OnError_ShouldRollbackConformanceValidationState()
```

Comment:
I don't quite understand what the "rollback" applies to in this test.

Reply:
This particular test surfaces a bug that existed in the checkpoint restore logic, which would surface as a false positive duplicate key conformance error. ([relevant fix](https://github.com/dotnet/runtime/pull/36770/files#diff-6767ae74ab4174c8585fb7fe1f2433e8R563-R568)). I've renamed it slightly and added more comments, though not sure what a good test name might be given it concerns implementation detail.